invenio_administration/assets/semantic-ui/js/invenio_administration/src/formik/fields/fields.js

```diff
@@ -0,0 +1,119 @@
+import { generateObjectFieldProps, ObjectField } from "./object";
+import { DynamicSubFormField, generateDynamicFieldProps } from "./dynamic";
+import { generateHiddenFieldProps } from "./hidden";
+import React from "react";
+import {
+  Input,
+  AutocompleteDropdown,
+  Dropdown,
+  TextArea,
+  RichInput,
+} from "react-invenio-forms";
+import { Field } from "formik";
+import _get from "lodash/get";
+import { AdminArrayField } from "./array";
+import _isEmpty from "lodash/isEmpty";
+import { sortFields } from "../../components/utils";
+import { AdminBoolField } from "./bool";
+import { generateFieldProps } from "./props_generator";
+import { generateVocabularyFieldProps } from "./vocabulary";
+import { generateArrayFieldProps } from "./array";
+
+const fieldsMap = {
+  string: { element: Input, props: generateFieldProps },
+  integer: { element: Input, props: generateFieldProps },
+  uuid: { element: Input, props: generateFieldProps },
+  datetime: { element: Input, props: generateFieldProps },
+  array: { element: AdminArrayField, props: generateArrayFieldProps },
+  bool: { element: AdminBoolField, props: generateFieldProps },
+  hidden: { element: Field, props: generateHiddenFieldProps },
+  vocabulary: { element: AutocompleteDropdown, props: generateVocabularyFieldProps },
+  dynamic: { element: DynamicSubFormField, props: generateDynamicFieldProps },
+  object: { element: ObjectField, props: generateObjectFieldProps },
+  dropdown: { element: Dropdown, props: generateFieldProps },
+  textarea: { element: TextArea, props: generateFieldProps },
+  html: { element: RichInput, props: generateFieldProps },
+  function: null,
+};
+
+export const mapFormFields = (
```

Comment:
it was moved from GenerateForm.js and refactored to improve readability. If anyone has an idea how to get rid of various `if` statements in the main function body, especially when trying to assign a correct type to a field, I would appreciate suggestions1